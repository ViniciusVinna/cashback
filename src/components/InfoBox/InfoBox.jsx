import React, { Fragment } from 'react';

import { InfoboxStyled } from './InfoBox.styled';

const {
  Item,
  List,
} = InfoboxStyled;

const InfoBox = () => (
  <Fragment>
    <List>
      <Item>
        <span>3%</span> de cashback em compras acima de R$ 50,00.
      </Item>

      <Item>
        <span>5%</span> de cashback em compras acima de R$ 250,00 atÃ© R$ 500,00.
      </Item>

      <Item>
        <span>8%</span> de cashback em compras a partir de R$ 500,00.
      </Item>

      <Item>
        <span role="img" aria-label="AtenÃ§Ã£o">ğŸ‘‰</span>Compras acima de R$ 500,00 sÃ£o validadas apÃ³s 2 dias Ãºteis.
      </Item>
    </List>
  </Fragment>
);
export default InfoBox;
