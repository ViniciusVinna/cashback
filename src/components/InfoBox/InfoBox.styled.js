import styled from '@emotion/styled';
import { css } from '@emotion/core';

export const InfoboxStyled = {
  List: styled('ul')(({ theme }) => css`
    background-color: ${theme.color.n300};
    border-radius: ${theme.borderRadius.m};
    color: ${theme.color.d400};
    font-size: ${theme.textSize.xs};
    line-height: ${theme.lineHeight.s};
    list-style: none;
    margin: ${theme.space.m} 0 0 0;
    padding: ${theme.space.m};
  `),
  Item: styled('li')(({ theme }) => css`
    border-bottom: 1px solid ${theme.color.d200};
    padding: ${theme.space.xs} 0;

    &:last-of-type {
      border: none;
    }

    & span {
      color: ${theme.color.d500};
      font-weight: ${theme.fontWeight.m};
    }
  `),
};
