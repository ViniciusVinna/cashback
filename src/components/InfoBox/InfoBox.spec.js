import React from 'react';
import InfoBox from './InfoBox';

describe('InfoBox', () => {
  it('it should render properly', () => {
    const { container } = render(<InfoBox />);

    expect(container).toMatchSnapshot();
  });
});
