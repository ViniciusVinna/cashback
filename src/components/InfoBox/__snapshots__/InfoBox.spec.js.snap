// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoBox it should render properly 1`] = `
<div>
  <ul
    class="css-1x9daa0"
  >
    <li
      class="css-zotes9"
    >
      <span>
        3%
      </span>
       de cashback em compras acima de R$ 50,00.
    </li>
    <li
      class="css-zotes9"
    >
      <span>
        5%
      </span>
       de cashback em compras acima de R$ 250,00 atÃ© R$ 500,00.
    </li>
    <li
      class="css-zotes9"
    >
      <span>
        8%
      </span>
       de cashback em compras a partir de R$ 500,00.
    </li>
    <li
      class="css-zotes9"
    >
      <span
        aria-label="AtenÃ§Ã£o"
        role="img"
      >
        ðŸ‘‰
      </span>
      Compras acima de R$ 500,00 sÃ£o validadas apÃ³s 2 dias Ãºteis.
    </li>
  </ul>
</div>
`;
